### ISSUE TITLE
Have couriers deliver stuff directly from a building to another

### ISSUE THREAD

**Issue by warning-djkajslh** [ldtteam/minecolonies#6885]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version:
- MineColonies Version:
- Structurize Version:
- Related Mods and their Versions:

## Expected behavior
Couriers will transfer item directly from building A to B without storing them into the warehouse first. It really bothering me knowing transfer between two buildings next to each other will need to go through the warehouse from far away.

## Actual behavior
Couriers will first take the stuff to put them into the warehouse far way first, then deliver the items to the building B that are a few blocks away from building A

## Steps to reproduce the issue

1. 
2. 
3. 

### Logs

- latest.log: 
- crashlog: 

### Notes

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
That's not true, from a crafter to a worker the delivery is direct when it was requested. When it's not a direct request fulfillment indirect delivery is done, but that is the case since we can only detect that on the warehouse dump well.
